src/lib.rs

```diff
@@ -584,8 +588,7 @@ fn create_file_paths(inputs: &[String]) -> Result<Vec<FilePath>, Diagnostic> {
 /// Links any provided libraries
 /// Returns the location of the output file
 pub fn build_with_params(parameters: CompileParameters) -> Result<(), Diagnostic> {
```

Comment:
I would diverge earlier, create the params from the subcommand and then call this method as before
So check for subcommand info already in main/cli